flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/VirtualChannelSelector.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.api;
+
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.runtime.event.RuntimeEvent;
+
+import java.util.Objects;
+
+/**
+ * An event that is used to demultiplex virtual channels over the same physical channel.
+ */
+public final class VirtualChannelSelector extends RuntimeEvent {
+
+	private final int subtaskIndex;
+	private final int channelIndex;
```

Comment:
The meaning of these fields is a bit ambiguous to me
- both: upstream or downstream
- channelIndex: subPartition or inputChannel

Some javadoc would be helpful.

Reply:
I changed it to the following to clarify that it's always meant to be from the perspective of the input side (where demultiplexing is happening).

```
/**
 * An event that is used to demultiplex virtual channels over the same physical channel. The information is to be interpreted from
 * the point of view of the downstream node.
 */
public final class VirtualChannelSelector extends RuntimeEvent {

	private final int inputSubtaskIndex;
	private final int inputChannelIndex;
```